lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -4078,6 +4255,9 @@ void RegisterRequestCallbacks() {
   g_dap.RegisterRequestCallback("threads", request_threads);
   g_dap.RegisterRequestCallback("variables", request_variables);
   g_dap.RegisterRequestCallback("disassemble", request_disassemble);
+  // Instruction breapoint request
```

Comment:
we also need to update the stopped event to use `reason: instruction breakpoint` instead of `reason: breakpoint` in case we hit one of those breakpoints

Reply:
@vogelsgesang 
Thanks for bringing this point.
The GDB remote protocol does not have a specific packet for instruction breakpoints, which is likely why LLDB manages instruction breakpoints using the z0 packet, just like it does for software breakpoints.
The challenge is distinguishing between instruction breakpoints and software breakpoints when they use the same packet type.

**Instruction breakpoints packet info:**
$[{"name":"mandelbrot","reason":"breakpoint","registers":{"16":"3046555555550000","6":"10e3ffffff7f0000","7":"00e3ffffff7f0000"}],"signal":5,"tid":12342}]]#72

**Source breakpoint packet info:**
$[{"name":"mandelbrot","reason":"breakpoint","registers":{"16":"2e46555555550000","6":"10e3ffffff7f0000","7":"00e3ffffff7f0000"}],"signal":5,"tid":12342}]]#a6

It would be helpful if you could share any insights on this.

Reply:
I don't really know about those differences on the gdb protocol. I was only reading https://microsoft.github.io/debug-adapter-protocol/specification#Requests_SetInstructionBreakpoints

> When an instruction breakpoint is hit, a stopped event (with reason `instruction breakpoint`) is generated.

Reply:
you could make something synthetic, and that could be very nice.
In the process event loop in lldb-dap, you could check if you stopped because of a breakpoint, and then you check if the breakpoint is an instruction breakpoint, and then you return that custom stop reason.
lldb-dap already does something like that in `CreateThreadStopped` for exception breakpoints.

Reply:
Thanks @walter-erquinigo 
This will work